pkg/api/admin/administration.go

```diff
@@ -402,25 +402,31 @@ func AssociateSecret(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	groupSecrets, rksErr := vaultClient.GetGroupSecretList(group)
-	if rksErr != nil {
-		rksErr.HandleErr(r.Context(), w)
-		return
-	}
+	var groupSecrets *model.GroupSecrets
 
-	for _, secret := range groupSecrets.Secrets {
-		if secret == fqdn {
-			(&model.RksError{WrappedError: nil, Message: fqdn + "already associated to " + group, Code: 409}).HandleErr(r.Context(), w)
+	for {
```

Comment:
As discussed yesterday, I'd put a max retry number in the for loop to avoid infinite loop in case Vault becomes unavailable. I'm not sure about the number of retry to allow. We should test different values with the locust test case

something like this:
```go
i := 0
for {
  // ...
  if i > 3 {
  // RksError 500
  }
  i++
}